src/main/telemetry/smartport.c

```diff
@@ -198,19 +262,29 @@ static void smartPortSendByte(uint8_t c, uint16_t *crcp)
     *crcp = crc;
 }
 
-static void smartPortSendPackage(uint16_t id, uint32_t val)
+static void smartPortSendPackageEx(uint8_t primId, uint8_t* data)
 {
     uint16_t crc = 0;
-    smartPortSendByte(FSSP_DATA_FRAME, &crc);
+    smartPortSendByte(primId, &crc);
+    for(uint8_t i=0;i<6;i++) {
```

Comment:
@ledvinap is correct. C compilers generate the most efficient code when local variables are the "natural size" of the machine, that is `int` or `unsigned int`. Betaflight derived from MultiiWii, a flight controller written for the Aruduino, so there are still plenty of local variables that are`uint8_t` - these are gradually being cleaned up. But certainly new code shouldn't use them for local variables.

[Note that this does not mean avoiding `unit8_t`s altogether - usage in `structs` and `static` variables is fine]